contracts/PetPark.sol

```diff
@@ -81,54 +69,58 @@ contract PetPark {
     }
 
     function borrow(
-        uint age,
+        uint _age,
         Gender _gender,
         AnimalType _animal
     )
         external
-        isOldEnough(age)
-        isValidAnimal2(_animal)
+        isOldEnough(_age)
+        isValidAnimalType(_animal)
         isAvailableAnimal(_animal)
     {
-        if (_gender == Gender.Male) {
-            if (_animal != AnimalType.Fish || _animal != AnimalType.Dog) {
-                revert("Invalid animal for men");
-            }
+        User storage u = userInfo[msg.sender];
+
+        if (u.age != _age && u.age != 0) {
+            revert("Invalid Age");
         }
 
-        if (_gender == Gender.Female) {
-            if (_animal == AnimalType.Cat && age < 40) {
-                revert("Invalid animal for women under 40");
-            }
+        if (u.gender != _gender && _gender == Gender.Female) {
```

Comment:
This is what is unclear to me. Zero value for enums is the first value, so the "default" gender is Male. I'm struggling to work that logic into the check for when a user hasn't borrowed any animals yet. This makes sense to me but it leads to the women <40 cat test failing.

Reply:
Instead of thinking of a negative test case, how would you write a condition for women under the age of 40 inside the if clause? And then lets if they are requesting a cat, if so then revert.